src/platforms/windows/daemon/windowsroutemonitor.cpp

```diff
@@ -68,6 +68,45 @@ WindowsRouteMonitor::~WindowsRouteMonitor() {
   logger.debug() << "WindowsRouteMonitor destroyed.";
 }
 
+void WindowsRouteMonitor::updateValidInterfaces(int family) {
+  PMIB_IPINTERFACE_TABLE table;
+  DWORD result = GetIpInterfaceTable(family, &table);
+  if (result != NO_ERROR) {
+    logger.warning() << "Failed to retrive interface table." << result;
+    return;
+  }
+
+  // Flush the list of interfaces that are valid for routing.
+  if ((family == AF_INET) || (family == AF_UNSPEC)) {
+    m_validInterfacesIpv4.clear();
+  }
+  if ((family == AF_INET6) || (family == AF_UNSPEC)) {
+    m_validInterfacesIpv6.clear();
+  }
+
+  // Rebuild the list of interfaces that are valid for routing.
+  for (ULONG i = 0; i < table->NumEntries; i++) {
+    MIB_IPINTERFACE_ROW* row = &table->Table[i];
+    if (row->InterfaceLuid.Value == m_luid) {
+      continue;
+    }
+    if (!row->Connected) {
+      continue;
+    }
+
+    if (row->Family == AF_INET) {
+      logger.debug() << "Interface" << row->InterfaceIndex << "is valid for IPv4 routing";
+      m_validInterfacesIpv4.append(row->InterfaceLuid.Value);
+    }
+    if (row->Family == AF_INET6) {
+      logger.debug() << "Interface" << row->InterfaceIndex << "is valid for IPv6 routing";
+      m_validInterfacesIpv6.append(row->InterfaceLuid.Value);
+    }
+  }
+
+  FreeMibTable(table);
```

Comment:
To be 100% sure we do not add a random `return` in this function, what about if we use a `qScopeGuard()` for this free?